Kitodo/src/main/java/org/kitodo/production/migration/TaskComparer.java

```diff
@@ -11,55 +11,57 @@
 
 package org.kitodo.production.migration;
 
-import java.util.Comparator;
 import java.util.Objects;
 
 import org.kitodo.data.database.beans.Task;
 
-public class TaskComparator implements Comparator<Task> {
+public class TaskComparer {
 
-    @Override
-    public int compare(Task firstTask, Task secondTask) {
+    /**
+     * Compares two tasks, if they are equal for migration purposes.
+     * @return true, if they are equal, false if not
+     */
+    public boolean isEqual(Task firstTask, Task secondTask) {
         if (Objects.isNull(firstTask) || Objects.isNull(secondTask)) {
-            return 1;
+            return false;
         }
         if (Objects.isNull(firstTask.getTitle()) ? Objects.nonNull(secondTask.getTitle())
                 : !firstTask.getTitle().equals(secondTask.getTitle())) {
-            return 1;
+            return false;
         }
         if (Objects.isNull(firstTask.getOrdering()) ? Objects.nonNull(secondTask.getOrdering())
                 : !firstTask.getOrdering().equals(secondTask.getOrdering())) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeAutomatic() != secondTask.isTypeAutomatic()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeMetadata() != secondTask.isTypeMetadata()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeImagesRead() != secondTask.isTypeImagesRead()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeImagesWrite() != secondTask.isTypeImagesWrite()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeExportDMS() != secondTask.isTypeExportDMS()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeAcceptClose() != secondTask.isTypeAcceptClose()) {
-            return 1;
+            return false;
         }
         if (firstTask.isTypeCloseVerify() != secondTask.isTypeCloseVerify()) {
-            return 1;
+            return false;
         }
         if (Objects.isNull(firstTask.getScriptPath()) ? Objects.nonNull(secondTask.getScriptPath())
                 : !firstTask.getScriptPath().equals(secondTask.getScriptPath())) {
-            return 1;
+            return false;
         }
         if (firstTask.isBatchStep() != secondTask.isBatchStep()) {
-            return 1;
+            return false;
         }
-        return 0;
+        return true;
```

Comment:
The last `if` condition an the `return true;` at the end can be simplified to 
`return firstTask.isBatchStep() == secondTask.isBatchStep();`